src/commonMain/kotlin/data/buffs/BadgeOfTheSwarmguard.kt

```diff
@@ -0,0 +1,81 @@
+package data.buffs
+
+import character.Ability
+import character.Buff
+import character.Proc
+import character.Stats
+import data.model.Item
+import sim.Event
+import sim.SimParticipant
+
+class BadgeOfTheSwarmguard : Buff() {
+    companion object {
+        const val name = "Badge of the Swarmguard (static)"
+    }
+    override val id: Int = 26480
+    override val name: String = Companion.name
+    override val durationMs: Int = -1
+    override val hidden: Boolean = true
+
+    val buffDurationMs = 30000
+    val buff = object : Buff() {
+        override val name: String  = "Badge of the Swarmguard"
+        override val durationMs: Int = buffDurationMs
+
+        val stackBuff = object : Buff() {
```

Comment:
We can still combine this inline buff with the containing one.  Currently, these two are redundant, and we should be able to remove `stackBuff`.  

We can rename L22 to "Insight of the Qiraji", add the `maxStacks` config to that Buff object.  Then, the state check on L60 can reference `this`'s state object, instead of `stackBuff`.  

The Kotlin type system will get a little confused, since the proc on L32 would be adding `this` Buff to the sim, but `this` technically doesn't exist yet since the Proc construction is contained within that same Buff object.  

Here is an example of how this can be worked around - just pass `this` to a function outside the `Buff` declaration that returns a `Proc`.  https://github.com/marisa-ashkandi/tbcsim/blob/main/src/commonMain/kotlin/character/classes/shaman/abilities/ElementalMastery.kt#L49